src/uwtools/config/tools.py

```diff
@@ -38,6 +38,43 @@ def compare_configs(
     return cfg_a.compare_config(cfg_b.data)
 
 
+def config_check_depths(
+    input_obj: Union[dict, Config], target_format: Optional[str], mode: Optional[str]
```

Comment:
What happens when `target_format` is `None`?

What happens if `mode` is `None`?

This seems like another opportunity to define a strings object instead of introducing magic strings.

Reply:
True - I've updated this to be required str, and perform a mode lookup check

Reply:
Just chucked the mode entry for separate functions as recommended!